### PR TITLE
Use pytest tmp_path fixture

### PR DESCRIPTION
**Pull request by daniel-thom**
## Description
I noticed that pytest provides a fixture to manage temporary directories. This PR removes the temporary directories that we were creating in our tests. No functional changes.

### CODE REVIEW

File: tests/test_filter_registry.py [hunk 1]
```diff
@@ -40,45 +36,42 @@
 }
 
 
-def test_filter_registry():
+def test_filter_registry(tmp_path):
```
Comment (lixiliu) [inline tests/test_filter_registry.py:39]: so this works because of the keyword `tmp_path`?

Comment (daniel-thom) [reply tests/test_filter_registry.py:39]: `tmp_path` is a built-in pytest fixture.
https://docs.pytest.org/en/7.1.x/how-to/tmp_path.html

Comment (lixiliu) [reply tests/test_filter_registry.py:39]: cool!

**Review by lixiliu — APPROVED** [review]
pytest passed for me

**Pull request merged**